types/wlr_output_layout.c

```diff
@@ -93,38 +103,68 @@ static struct wlr_box *output_layout_output_get_box(
  * the rightmost output in the layout in a horizontal line.
  */
 static void output_layout_reconfigure(struct wlr_output_layout *layout) {
-	int max_x = INT_MIN;
-	int max_x_y = INT_MIN; // y value for the max_x output
 
-	// find the rightmost x coordinate occupied by a manually configured output
-	// in the layout
 	struct wlr_output_layout_output *l_output;
+
 	wl_list_for_each(l_output, &layout->outputs, link) {
-		if (l_output->state->auto_configured) {
-			continue;
-		}
+		int max_x = INT_MIN;
+		int max_x_y = INT_MIN;
 
 		struct wlr_box *box = output_layout_output_get_box(l_output);
-		if (box->x + box->width > max_x) {
-			max_x = box->x + box->width;
-			max_x_y = box->y;
-		}
-	}
 
-	if (max_x == INT_MIN) {
-		// there are no manually configured outputs
-		max_x = 0;
-		max_x_y = 0;
-	}
+		switch(l_output->configuration) {
+		case WLR_OUTPUT_LAYOUT_OUTPUT_CONFIGURATION_FIXED:
+			break;
+		case WLR_OUTPUT_LAYOUT_OUTPUT_CONFIGURATION_RELATIVE_LEFT_OF: {
+				struct wlr_box *ref_box =
+					output_layout_output_get_box(l_output->reference);
```

Comment:
This introduces dependencies between configurations. It means that if the reference is for some reason configured after this output, this output won't get configured properly.